pkg/controllers/provisioning/static/controller.go

```diff
@@ -0,0 +1,146 @@
+	if !c.cluster.Synced(ctx) {
+		return reconcile.Result{RequeueAfter: time.Second}, nil
+	}
+	if !nodepoolutils.IsManaged(np, c.cloudProvider) {
+		return reconcile.Result{}, nil
+	}
+	if !np.StatusConditions().Root().IsTrue() {
+		return reconcile.Result{}, nil
+	}
+	if np.Spec.Replicas == nil {
+		return reconcile.Result{}, nil
+	}
+	nodes := c.cluster.NodePoolResourcesFor(np.Name)[resources.Node]
+	// Size down of replicas will be handled in disruption controller to drain nodes and delete NodeClaims
+	if nodes.Value() > lo.FromPtr(np.Spec.Replicas) {
+		return reconcile.Result{}, nil
+	}
+	// Create the number of NodeClaims equal to the desired replica count
+	var nodeClaims []*scheduling.NodeClaim
+	for range lo.FromPtr(np.Spec.Replicas) - nodes.Value() {
+		// Need fresh copy everytime else nodeclaims would share the template resources causing races
+		nct := scheduling.NewNodeClaimTemplate(np)
+		nodeClaims = append(nodeClaims, &scheduling.NodeClaim{NodeClaimTemplate: *nct, IsStaticNode: true})
+	}
+	if _, err := c.ncProvisioner.CreateNodeClaims(ctx, nodeClaims, provisioning.WithReason(metrics.ProvisionedReason)); err != nil {
+		return reconcile.Result{}, fmt.Errorf("creating nodeclaims, %w", err)
+	}
+	return reconcile.Result{}, nil
+}
+
+func (c *Controller) Register(_ context.Context, m manager.Manager) error {
+	return controllerruntime.NewControllerManagedBy(m).
+		Named("provisioning.static").
+		For(&v1.NodePool{}, builder.WithPredicates(nodepoolutils.IsManagedPredicateFuncs(c.cloudProvider), predicate.Funcs{
+			CreateFunc: func(e event.CreateEvent) bool {
+				return true
+			},
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				oldNP := e.ObjectOld.(*v1.NodePool)
+				newNP := e.ObjectNew.(*v1.NodePool)
+				return HasNodePoolReplicaOrStatusChanged(oldNP, newNP)
+			},
+			DeleteFunc: func(e event.DeleteEvent) bool {
+				return false
+			},
+			GenericFunc: func(e event.GenericEvent) bool {
+				return false
+			},
+		})).
+		Watches(&v1.NodeClaim{}, nodeutils.NodeClaimEventHandler(c.kubeClient), builder.WithPredicates(predicate.Funcs{
+			CreateFunc: func(e event.CreateEvent) bool {
+				return true
+			},
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				return e.ObjectOld.GetDeletionTimestamp().IsZero() && !e.ObjectNew.GetDeletionTimestamp().IsZero()
+			},
+			DeleteFunc: func(e event.DeleteEvent) bool {
+				return true
+			},
+			GenericFunc: func(e event.GenericEvent) bool {
+				return false
+			},
+		})).
+		WithOptions(controller.Options{MaxConcurrentReconciles: 10}).
+		Complete(reconcile.AsReconciler(m.GetClient(), c))
+}
+
+func HasNodePoolReplicaOrStatusChanged(oldNP, newNP *v1.NodePool) bool {
+	if oldNP.Spec.Replicas != newNP.Spec.Replicas {
+		return true
+	}
+	oldReady := oldNP.StatusConditions().Root().IsTrue()
+	newReady := newNP.StatusConditions().Root().IsTrue()
+	return oldReady != newReady
```

Comment:
We really only want to reconcile here when the new status transitions to True and the old one was something other than true, not just when they are equal

Reply:
Was checking if new has changed, will make a change to trigger reconcilation only when we transition to true from something other than true